pilot/pkg/xds/gen2.go

```diff
@@ -61,76 +62,30 @@ func init() {
 	controlPlane = &corev3.ControlPlane{Identifier: string(byVersion)}
 }
 
-// handleCustomGenerator uses model.Generator to generate the response.
-func (s *DiscoveryServer) handleCustomGenerator(con *Connection, req *discovery.DiscoveryRequest) error {
-	if !s.shouldRespond(con, nil, req) {
-		return nil
-	}
-
-	push := s.globalPushContext()
-	resp := &discovery.DiscoveryResponse{
-		ControlPlane: ControlPlane(),
-		TypeUrl:      req.TypeUrl,
-		VersionInfo:  push.Version, // TODO: we can now generate per-type version !
-		Nonce:        nonce(push.Version),
-	}
-	if push.Version == "" { // Usually in tests.
-		resp.VersionInfo = resp.Nonce
-	}
-
-	// XdsResourceGenerator is the default generator for this connection. We want to allow
-	// some types to use custom generators - for example EDS.
-	g := con.proxy.XdsResourceGenerator
-	if cg, f := s.Generators[con.proxy.Metadata.Generator+"/"+req.TypeUrl]; f {
-		g = cg
-	}
-	if cg, f := s.Generators[req.TypeUrl]; f {
-		g = cg
-	}
-	if g == nil {
-		g = s.Generators["api"] // default to MCS generators - any type supported by store
-	}
-
-	if g == nil {
-		return nil
-	}
-
-	cl := g.Generate(con.proxy, push, con.Watched(req.TypeUrl), nil)
-	sz := 0
-	for _, rc := range cl {
-		resp.Resources = append(resp.Resources, rc)
-		sz += len(rc.Value)
-	}
-
-	err := con.send(resp)
-	if err != nil {
-		recordSendError("ADS", con.ConID, apiSendErrPushes, err)
-		return err
-	}
-	apiPushes.Increment()
-
-	adsLog.Infof("%s: PUSH for node:%s resources:%d", v3.GetShortType(req.TypeUrl), con.proxy.ID, len(cl))
-
-	return nil
+var SkipLogTypes = map[string]struct{}{
+	v3.EndpointType: {},
 }
 
-// TODO: verify that ProxyNeedsPush works correctly for Generator - ie. Sidecar visibility
-// is respected for arbitrary resource types.
-
 // Called for config updates.
 // Will not be called if ProxyNeedsPush returns false - ie. if the update
-func (s *DiscoveryServer) pushGeneratorV2(con *Connection, push *model.PushContext,
-	currentVersion string, w *model.WatchedResource, updates model.XdsUpdates) error {
-	gen := s.Generators[w.TypeUrl]
+func (s *DiscoveryServer) pushGenerator(con *Connection, push *model.PushContext,
```

Comment:
Should we call pushGenerator as pushXds? Generator is used to Generate Xds Resources but ultimately we are pushing xds resources?